Libraries/LibGUI/ModelSelection.cpp

```diff
@@ -38,8 +38,14 @@ void ModelSelection::remove_matching(Function<bool(const ModelIndex&)> filter)
         if (filter(index))
             to_remove.append(index);
     }
-    for (auto& index : to_remove)
-        m_indexes.remove(index);
+    if (!to_remove.is_empty()) {
+        for (auto& index : to_remove)
+            m_indexes.remove(index);
+        if (!m_disable_notify)
+            notify_selection_changed();
+        else
+            m_notify_pending = true;
```

Comment:
This exact code is repeated six times. I think it would make more sense to move that logic inside the `notify_selection_changed` function (and expect that function to just set the pending flag if notifications are disabled).